tools/plot-bench.py

```diff
@@ -0,0 +1,305 @@
+        self.ax.set_ylabel('Runtime at the $1 + i$th index')
+        self.ax.scatter(x, y, marker='.')
+        self.ax.grid()
+
+
+def plotters_factory(dpi: Any = 100, transparent: bool = False) -> Dict[str, Plotter]:
+    
+    return {
+        'lag-plot': lambda : LagPlot(dpi=dpi, transparent=transparent),
+        'runtime-densities': lambda: RuntimeDensities(dpi=dpi, transparent=transparent),
+        'cumsum-per-run': lambda: CumsumPerRun(dpi=dpi, transparent=transparent),
+        'per-run': lambda: PerRun(dpi=dpi, transparent=transparent)
+    }
+
+
+def chunks(data: Dict, size: int):
+    """Generator that makes sub-dictionaries of a maximum size."""
+    it = iter(data)
+    for _ in range(0, len(data), size):
+        yield {k:data[k] for k in islice(it, size)}
+
+
+def get_args() -> Any:
+    """Gets the arguments used in the program."""
+    parser = argparse.ArgumentParser(
+        prog='Futhark Plots',
+        description='Makes plots for futhark benchmarks.'
+    )
+    parser.add_argument('filename', metavar='FILE',
+        help='the benchmark results as a json file generated by futhark bench.')
+    parser.add_argument('--programs', metavar='PROGRAM0,PROGRAM1,...',
+        help='the specific programs the plots will be generated from.')
+    parser.add_argument('--plots', metavar="PLOTTYPE0,PLOTTYPE1,...",
+        help=(f'the type of plots that will be generated which can be '
+              f'{", ".join(plotters_factory().keys())}.'))
+    parser.add_argument('--backend', default='pdf', metavar='BACKEND',
+        help='the type of backend used, these can be found on the matplotlib website.')
+    parser.add_argument('--filetype', default='png',  metavar='BACKEND',
+        help='the file type used, these can be found on the matplotlib website.')
+    parser.add_argument('--quality', default='low', metavar='QUALITY',
+        choices=['low', 'medium', 'high'], help='The quality of the plot')
+    parser.add_argument('--transparent', action='store_true',
+        help='flag to use if the bagground should be transparent.')
+
+    return parser.parse_args()
+
+
+def format_arg_list(args: Optional[str]) -> Optional[str]:
+    """
+    Takes a string of form 'a, b, c, d' and makes a list ['a', 'b', 'c', 'd']
+    """
+    if args is None:
+        return None
+    
+    return set(map(lambda arg: arg.strip(), args.split(',')))
+
+
+def make_plot_jobs_and_directories(
+    programs: Set[str],
+    plot_jobs: Dict[str, Dict[str, Dict[str, Any]]],
+    data: Dict[str, Dict[str, Dict[str, Any]]],
+    plots: Set[str],
+    filetype: str
+    ) -> None:
+    """Makes dictionary with plot jobs where plot_jobs are the jobs."""
+
+    for program in programs:
+        datasets = data.get(program).get('datasets')
+        for dataset_path, dataset in datasets.items():
+            runtimes = dataset.get('runtimes')
+            directory = os.path.dirname(program)
+            program_basename = pathlib.Path(program).stem
```

Comment:
This is wrong, as it cuts off the entry point name in things like `"futhark-benchmarks/babelstream/babelstream.fut:f32_add"`.

Reply:
This should be correct now but resulted in more rendering time.